qiskit/extensions/quantum_initializer/initializer.py

```diff
@@ -99,6 +110,27 @@ def _define_fromlabel(self):
 
         return initialize_circuit
 
+    def _define_from_int(self):
+        q = QuantumRegister(self.num_qubits, 'q')
+        initialize_circuit = QuantumCircuit(q, name='init_def')
+
+        # Convert to int since QuantumCircuit converted to complex
+        # and make a bit string and reverse it
+        intstr = f'{int(np.real(self.params[0])):0b}'[::-1]
```

Comment:
This doesn't apply the reset gate on all qubits right now (but it should):
```
>>> qc = QuantumCircuit(3)
>>> qc.initialize(1, [0, 1, 2])
>>> qc.draw()
     ┌────────────────┐
q_0: ┤0               ├
     │                │
q_1: ┤1 initialize(1) ├
     │                │
q_2: ┤2               ├
     └────────────────┘
>>> qc.decompose().draw()
          ┌───┐
q_0: ─|0>─┤ X ├
          └───┘
q_1: ──────────

q_2: ──────────
```

This is likely fixed by expanding the bitstring onto all qubits:
```suggestion
        param = int(np.real(self.params[0]))
        intstr = bin(param)[2:].zfill(self.num_qubits)[::-1]
```

Reply:
Ah, good catch. I originally had `intstr = f'{int(np.real(self.params[0])):0{self.num_qubits}b}'[::-1]`, but lost the num_qubits somehow.

Done in https://github.com/Qiskit/qiskit-terra/pull/5418/commits/8a7f33f292af59659e661aff949c9e31a25ec79a